zerver/lib/actions.py

```diff
@@ -6307,7 +6307,8 @@ def do_update_message(
         assert target_message.is_stream_message()
         assert stream_being_edited is not None
 
-        edit_history_event["prev_stream"] = stream_being_edited.id
+        edit_history_event["prev_stream"] = stream_being_edited.name
+        edit_history_event["new_stream"] = new_stream.name
```

Comment:
Changing this this format won't work, since the new field won't be present for any existing events in the database.  

We should be able to get the stream name from its ID using a `stream_data` or `sub_store.js` lookup from the stream ID, which I believe should be present wherever relevant.

Reply:
(Though we should catch the exception if the related stream is unknown, and display "Unknown stream" for that case -- this can happen if e.g. content was moved from a private stream the current user doesn't have access to).

Reply:
Yeah! I have now used `stream_data` to get the names through their id. But the `edit_history_event` for `new_stream` is required, I tried many ways out but I guess this resulted in the correct changes required for the issue. Kindly correct me if I'm wrong.

Reply:
I still don't think this change is required; the new stream is always available as either the the `prev_stream` of the next edit history event that changes the stream, 

That said, @cyphase had a project that would make the API do this chaining work for you in the server, so the frontend doesn't have to, and I think that's what we want here: https://github.com/zulip/zulip/pull/17193.